interfaces/builtin/opencl.go

```diff
@@ -0,0 +1,53 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/ubuntu-core/snappy/interfaces"
+)
+
+const openclConnectedPlugAppArmor = `
+# Description: Can access opencl. 
+# Usage: reserved
+
+  # OpenCL Installable Client Driver (ICD) Loader
+  /etc/OpenCL/vendors/** r,
```

Comment:
I'm trying to introduce the good and bad about OpenCL below.  
- OpenCL is (basically) closed source stuff. /etc/OpenCL/vendors/ is the place where vendors save _information_ about OpenCL SDK availability. 
   - If you have none GPU with proper drivers, the /etc/OpenCL/vendors dir doens't exist (if possible, a check as proposed by @niemeyer will be an improvement).
   - E.g., when you installed AMD drivers+SDK the directory is created with **information**. See below.

```
$ cd /etc/OpenCL/vendors/

$ ls
amdocl32.icd  amdocl64.icd  intel64.icd  nvidia.icd

$ cat amdocl64.icd 
libamdocl64.so

$ whereis libamdocl64.so
libamdocl64: /usr/lib64/libamdocl64.so
```
1. all important stuff is closed source; can't be inside the snap for a real use case. 
2. the reference implementation (ocl-icd-libopencl1 ubuntu package) handle what I did below: check the dir, find the real lib, load, etc. 

I'm not sure how this possibly fits your view of a snap package. I'm ok with any complains, let's find the best path to follow.
- OpenCL is an open framework closely related to OpenGL. Have it 'supported' make sense.
- As you can see above, the real libraries are somewhere (I guess this will be a problem).
- OpenCL programs can fullfill a GPU with a high demand task (no responsive computer - if bad design program uses it). It is not hard to hang a computer (blame GPU vendors).
- When I link against ocl-icd-libopencl1 or use dlopen("libOpenCL.so", RTLD_LAZY)] ALL details are transparent for me, the developer (number 2 above).
- OpenCL is mature and has a future. I can show you a use case where using your fancy CPU and AVX/XOP/SSE you can do something at 1000 things per second. Using a GPU you can get 10000 - 30000 things per second.